chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -216,23 +188,23 @@ pub struct PeerManagerActor {
     txns_since_last_block: Arc<AtomicUsize>,
     /// Number of active peers, used for rate limiting.
     peer_counter: Arc<AtomicUsize>,
-    /// Used for testing, for disabling features.
-    adv_helper: AdvHelper,
     /// Whitelisted nodes, which are allowed to connect even if the connection limit has been
     /// reached.
     whitelist_nodes: Vec<WhitelistNode>,
-    /// test-only, defaults to () (a noop counter).
-    ping_counter: Box<dyn PingCounter>,
+    /// test-only.
+    event_sink: Sink<Event>,
```

Comment:
:thinking: so it seems for all events except for `RoutingTable` you could use `tracing` instead of explicit sink. Gathering tracing events, if not abused too much, seems like a fine way to get a bit more visibility than the public API allows.

Though, maybe with https://docs.rs/valuable/latest/valuable/trait.Valuable.html even routing table can be observed in this way?


Haven't reviewed the code in details, but, all things being equal, I would prefer using tracing over defining sink. But I can easily see how, for one reason or other, we need Sink!

Reply:
I've never heard about using tracing/Valuable in such context. Would you like to talk it through? I'll schedule something tomorrow.

Reply:
As I understand "tracing" adds logging structure, by providing a tree of spans, but the "field data" it collects is a [string -> string] map, is that correct? I suspect that Valuable is supposed some reflection mechanism, but I do not follow how does it fit into this case.

Right now I have:
- ServerStart event, which can be entirely moved out of PeerManager lifecycle, by passing an open port to constructor.
- Ping/Pong events which carries nonce and source
- RoutingTable which carries a map PeerId -> [PeerId]

I suspect that this is all the data I need to migrate integration tests to near_network. What I like about this setup is:
- it is strongly typed
- it is not global (can you subscribe to a specific span, or you have to actively filter out everything else in the subscriber? In particular with 2 instances of PeerManagerActor will I be able to distinguish events coming from each one?)
- it allows me to replace polling with await calls (once we replace actix run_later calls with await on a fake clock there should be no waiting in the test runs whatsoever).
Sink is just like a Subscriber, except it accepts typed data. A custom subscriber in practice will just be a wrapper on what I've already implemented, as I need some translation to async event anyway.

Reply:
closing, staying with Sink.